### PR TITLE
Update CI workflows pull_request path triggers

### PR DESCRIPTION
**Pull request by folmos-at-orange**

### CODE REVIEW

File: .github/workflows/pip.yml [hunk 1]
```diff
@@ -12,6 +12,7 @@ on:
         default: 10.2.3.0
         description: Development Docker Image Tag
   pull_request:
+    paths: [setup.py, versioneer.py, .github/workflows/pip.yml]
```
Comment (popescu-v) [inline .github/workflows/pip.yml:15]: IMHO, we should also have the following files in this list:
- `khiops/_version.py` (generated by `versioneer.py`),
- `pyproject.toml`,
- `setup.cfg`
And perhaps also `LICENSE.md`.

File: .github/workflows/conda.yml [hunk 2]
```diff
@@ -22,7 +22,12 @@ on:
   push:
     tags: ['*']
   pull_request:
-    paths: [packaging/conda/**, '!packaging/conda/README.md', .github/workflows/conda.yml]
+    paths:
+      - setup.py
+      - versioneer.py
+      - packaging/conda/**,
+      - '!packaging/conda/README.md'
+      - .github/workflows/conda.yml
```
Comment (popescu-v) [inline .github/workflows/conda.yml:30]: IMHO, all the files included for `pip.yml` should also be here (see the next comment). And also `LICENSE.md` IMHO.

File: .github/workflows/pip.yml (hunk 1 above)
Comment (folmos-at-orange) [reply .github/workflows/pip.yml:15]: Done

**Review by popescu-v — APPROVED** [review]
LGTM (squash merge).

**Pull request merged**